app/src/main/java/com/iic/shopingo/ui/HomeActivity.java

```diff
@@ -38,8 +37,19 @@ public boolean onOptionsItemSelected(MenuItem item) {
     //noinspection SimplifiableIfStatement
     if (id == R.id.action_settings) {
       return true;
+    } else if (id == R.id.action_logout) {
+      SharedUserConnector.getInstance().logout(this);
+      navigateToOnboarding();
+    } else if (id == R.id.action_update_contact) {
+      Intent intent = new Intent(this, ContactDetailsActivity.class);
+      startActivity(intent);
     }
 
     return super.onOptionsItemSelected(item);
   }
+
+  private void navigateToOnboarding() {
+    Intent intent = new Intent(this, OnboardingActivity.class);
+    startActivity(intent);
```

Comment:
Nope, when I try calling finish if caused issues.
But that's ok, since in the contact details activity when I navgiate to home activity I set the FLAG_ACTIVITY_CLEAR_TOP. So after arriving to home screen if I tap back then I return the app launcher.